packages/scene-composer/src/components/panels/scene-components/DataBindingComponentEditor.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useCallback, useContext } from 'react';
+import { SpaceBetween } from '@awsui/components-react';
+
+import { IComponentEditorProps } from '../ComponentEditor';
+import { ISceneComponentInternal, useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { IDataBindingComponentInternal } from '../../../store/internalInterfaces';
+
+import { DataBindingMapEditor } from './common/DataBindingMapEditor';
+
+export interface IDataBindingComponentEditorProps extends IComponentEditorProps {
+  component: IDataBindingComponentInternal;
+}
+
+export const DataBindingComponentEditor: React.FC<IDataBindingComponentEditorProps> = ({
+  node,
+  component,
+}: IDataBindingComponentEditorProps) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const removeComponent = useStore(sceneComposerId)((state) => state.removeComponent);
+  const valueDataBindingProvider = useStore(sceneComposerId)(
+    (state) => state.getEditorConfig().valueDataBindingProvider,
+  );
+
+  const onUpdateCallback = useCallback(
+    (componentPartial: Partial<IDataBindingComponentInternal>, replace?: boolean) => {
+      const componentPartialWithRef = { ref: component.ref, ...componentPartial };
+      if (componentPartialWithRef.valueDataBindings && componentPartialWithRef.valueDataBindings.length > 0) {
+        updateComponentInternal(node.ref, componentPartialWithRef as ISceneComponentInternal, replace);
```

Comment:
Nit: Consider how to avoid type assertion.

Reply:
done